app/models/project.rb

```diff
@@ -202,7 +206,8 @@ def self.selectable_projects
   # to everybody having at least one role in a project regardless of the
   # role's permissions.
   def self.visible_by(user = User.current)
-    allowed_to(user, :view_project)
+    # TODO: Temporary ...
+    allowed_to(user, :view_project).or(where(id: WorkPackage.visible(user).select(:project_id)))
```

Comment:
This will likely be generalized to allow visibility on the project or any entity visibility. Do you have plans for that? Is there a ticket already? I'm not a huge fan of a TODO that doesn't state why it is one.

Reply:
Removed the TODO and added a comment to https://community.openproject.org/projects/openproject/work_packages/51267 to see wether we need to revisit the scope